src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3RequestStream.cs

```diff
@@ -405,6 +419,8 @@ private async Task SendContentAsync(HttpContent content, CancellationToken cance
                 _stream.CompleteWrites();
             }
 
+            _requestSendCompleted = true;
+            RemoveFromConnectionIfDone();
```

Comment:
What happens in exceptional cases? Is it handled somehow differently? Same question for `_responseRecvCompleted`.

Reply:
_responseRecvCompleted is set to true in `HandleReadResponseContentException` so it is set in exceptional cases too.

I moved _requestSendCompleted to a finally block to make sure it covers both successful and unsuccessful paths.